ast-nodes/primary.dart

```diff
@@ -1,9 +1,46 @@
 import 'product.dart';
 import '../lexer.dart';
-
+import '../syntax-error.dart';
+import 'node.dart';
+import '../iterator-utils.dart';
+import 'boolean-literal.dart';
+import 'integer-literal.dart';
+import 'real-literal.dart';
+import 'modifiable-primary.dart';
 /// An abstract value.
 abstract class Primary implements Product {
   factory Primary.parse(Iterable<Token> tokens) {
+    if (tokens.isEmpty) {
+      throw SyntaxError(tokens.first, 'Expected a primary');
+    }
+    var iter = tokens.iterator;
+    iter.moveNext();
+    if (iter.current.value == 'true') {
+      checkNoMore(iter);
+      var bLBuffer = new BooleanLiteral(true)
+      return bLBuffer;
+
+    } else if (iter.current.value == 'false') {
+      checkNoMore(iter);
+      var bLBuffer = new BooleanLiteral(false)
+      return bLBuffer;
+
+    } else {
+      var tempToken = consumeUntil(iter, RegExp('\D'));
```

Comment:
And on the contrary, all integer and real literals are single tokens, so "153" and "1.56" are recognized as single tokens.